plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+    returned: when tag_flag is defined
+    type: int
+    sample: 819
+allow_uids:
+    description: Whether the SETUID and SETGID mode bits on executables in this file system are considered.
+    returned: always
+    type: bool
+    sample: yes
+sysname:
+    description: SYSNAME specifies the particular system on which a mount should be performed.
+    returned: if Non-None
+    type: str
+    sample: MVSSYS01
+automove:
+    description:
+        - >
+          Specifies what is to happens to the ownership of a file system during
+          a shutdown, PFS termination, dead system takeover, or file system move occurs.
+    returned: if Non-None
+    type: str
+    sample: AUTOMOVE
+automove_list:
+    description: This specifies the list of servers to include or exclude as destinations.
+    returned: if Non-None
+    type: str
+    sample: I,SERV01,SERV02,SERV03,SERV04
+msg:
+    description: Failure message returned by the module.
+    returned: failure
+    type: str
+    sample: Error while gathering information
+stdout:
+    description: The stdout from the tso mount command.
+    returned: always
+    type: str
+    sample: Copying local file /tmp/foo/src to remote path /tmp/foo/dest
+stderr:
+    description: The stderr from the tso mount command.
+    returned: failure
+    type: str
+    sample: No such file or directory "/tmp/foo"
+stdout_lines:
+    description: List of strings containing individual lines from stdout.
+    returned: failure
+    type: list
+    sample: [u"Copying local file /tmp/foo/src to remote path /tmp/foo/dest.."]
+stderr_lines:
+    description: List of strings containing individual lines from stderr.
+    returned: failure
+    type: list
+    sample: [u"FileNotFoundError: No such file or directory '/tmp/foo'"]
+cmd:
+    description: The actual tso command that was attempted.
+    returned: failure
+    type: str
+    sample: MOUNT EXAMPLE.DATA.SET /u/omvsadm/sample 3380
+rc:
+    description: The return code of the mount command, if applicable.
+    returned: failure
+    type: int
+    sample: 8
+
+"""
+
+import glob
+import math
+import os
+import re
+import stat
+import shutil
+import time
+import tempfile
+
+from datetime import datetime
+from pathlib import Path
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils.ansible_module import (
+    AnsibleModuleHelper,
+)
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils import (
+    better_arg_parser,
+    data_set,
+    encode,
+    vtoc,
+    #    backup,
+    copy,
+    mvs_cmd,
+)
+from ansible_collections.ibm.ibm_zos_core.plugins.module_utils.import_handler import (
+    MissingZOAUImport,
+)
+
+try:
+    from zoautil_py import Datasets, MVSCmd, types
+except Exception:
+    Datasets = MissingZOAUImport()
+    MVSCmd = MissingZOAUImport()
+    types = MissingZOAUImport()
+
+
+def cleanup(src_list):
+    pass
+
+
+def swap_text(original, adding, removing):
```

Comment:
Can swap_text remove persistent mounts that were not created by zos_mount?

Reply:
Yes.  That is the intention.  If it finds a match for mount and source, it will roll UP to the next non-comment or blank line, then roll through the mount command to the next non-footer /non-blank line, and remove the block found.